extensions/amp-loader/0.1/amp-loader.js

```diff
@@ -47,6 +46,38 @@ const LOADER_BACKGROUND_TAGS = {
   'AMP-GOOGLE-DOCUMENT-EMBED': true,
 };
 
+/**
+ * All video player components must either have a) "video" or b) "player" in
+ * their name.
+ * @private @const {!RegExp}
+ */
+const VIDEO_PLAYER_TAG_PATTERN = /\-(video|player)/i;
```

Comment:
FWIW, this regex also matches `amp-story-player`, which is not a video player.

/cc @Enriqe @gmajoulet

Reply:
This has historically been the case. Its effect is that it draws a specific background color behind the placeholder.

Reply:
Makes sense, but @Enriqe @gmajoulet are we aware of this?

> Its effect is that it draws a specific background color behind the placeholder.

I think my concern is that if we don't actually desire this semantically (regardless of what the specific effects are today), we'll get "surprised" when loaders have a requirements change and they "accidentally" get applied to `amp-story-player` as well.

Of course, this is totally fine if we just actually want this to apply to `amp-story-player` too, I just don't know.

Reply:
Can we update the regex and add a negative lookahead to exclude `story-player`?

Reply:
Done, but please take note:

- You were seeing a loader on `amp-story-player` previously because it would match this regex, it was not listed.
- You will **still** see a loader after excluding it from this regex, but you won't see the background for which there's a list.

- After this change, you can't opt-out of loaders: if you're sized (e.g. not `layout=container`) and you resolve an async `layoutCallback` after `600ms` you load, that's that.
- Loader logic is likely to be simplified

Reply:
@Enriqe can you take a look?

Reply:
Sorry, this got lost in my notifications.

Thanks @alanorozco for the context and additional information. For context, the player has its own loader that will eventually listen to a story-specific signal to stop displaying (see https://github.com/ampproject/amphtml/issues/32436). I think we can just use `this.toggleLoading(false)` like you mentioned to disable the default AMP loader.

For now, having this updated regex that excludes the `amp-story-player` from having the loader background color SGTM. 

> (Possibly, later in the future) Loader logic is likely to be simplified 

Could you please make sure to cc `@ampproject/wg-stories` so that we are aware of these future changes too? Thanks!